segmentstore/server/src/test/java/io/pravega/segmentstore/server/logs/ThrottlerTests.java

```diff
@@ -349,6 +349,37 @@ private void testInterruptedDelay(ThrottlerCalculator.ThrottlerName throttlerNam
         }
     }
 
+    /**
+     * Test the throttleOnce method to check its behavior when throttler is suspended
+     */
+    @Test
+    public void testThrottleOnceWhenSuspended() {
+        val delays = new ArrayList<>();
+        val calculator = new TestCalculatorThrottler(ThrottlerCalculator.ThrottlerName.Cache);
+        calculator.setDelayMillis(10000);
+        val suspended = new AtomicBoolean(true);
+        TestThrottler t = new TestThrottler(this.containerId, wrap(calculator), suspended::get, executorService(), metrics, delays::add);
+        t.throttleOnce(wrap(calculator).getThrottlingDelay());
+        Assert.assertTrue("Error: Throttler has continued to throttle inspite of there being System-Critical to process", t.lastDelayFuture.get().isCompletedExceptionally());
+    }
+
+    /**
+     * Test the throttleOnce method to check its behavior when throttler is not suspended
+     */
+    @Test
+    public void testThrottleOnceWhenNotSuspended() throws Exception {
+        val delays = new ArrayList<>();
+        val calculator = new TestCalculatorThrottler(ThrottlerCalculator.ThrottlerName.Cache);
+        int delayMillis = 5000;
+        calculator.setDelayMillis(delayMillis);
+        val suspended = new AtomicBoolean(false);
+        TestThrottler t = new TestThrottler(this.containerId, wrap(calculator), suspended::get, executorService(), metrics, delays::add);
+        CompletableFuture<Void> throt = t.throttleOnce(wrap(calculator).getThrottlingDelay());
+        TestUtils.await(throt::isDone, 5, TIMEOUT_MILLIS);
```

Comment:
Is this going to be waiting for 5 seconds? If so, this may induce an unnecessary long execution time for this test. Can we check that the future is not interrupted/cancelled instead of waiting for it to complete? If not, please significantly reduce the delay time (100ms or similar).

Reply:
have set this to 500 ms.